mautic/mautic
# Email domain field

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [x] 
| Related user documentation PR URL      | /
| Related developer documentation PR URL | /
| Issue(s) addressed                     | /



#### Description:

This PR creates a generated column to hold the domain portion of a contacts email address. Email domains are used very frequently by marketers for segment filtering. Yet, it is very slow to use `like`, `regex` or `endsWith` operators because those segment queries then do not use indexes.

This PR will create the new field automatically for all existing and new contacts and keeps the values up to date if the email address changes on the database level.

Marketers, please use the `Email Domain` field for segment filters. It will build the segments much faster and use less resources to do so.

#### Steps to test this PR:


1. Load up [this PR](https://mautibox.com)
2. Or apply this PR on your instance and run `bin/console doctrine:migrations:migrate` to add the generated field.
3. Create new or update an existing segment and you will be able to see the new **Email Domain** filter.
4. Use it in a segment with various operator and ensure that the right contacts are in the segment.


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10583"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
